spec/darlingtonia/csv_format_validator_spec.rb

```diff
@@ -3,5 +3,36 @@
 require 'spec_helper'
 
 describe Darlingtonia::CsvFormatValidator do
-  it_behaves_like 'a Darlingtonia::Validator'
+  subject(:validator)  { described_class.new }
+  let(:invalid_parser) { Darlingtonia::CsvParser.new(file: invalid_file) }
+  let(:invalid_file)   { File.open('spec/fixtures/bad_example.csv') }
+
+  it_behaves_like 'a Darlingtonia::Validator' do
+    let(:valid_parser)   { Darlingtonia::CsvParser.new(file: valid_file) }
+    let(:valid_file)     { File.open('spec/fixtures/example.csv') }
+  end
+
+  define :a_validator_error do
+    match do |error|
+      return false unless error.respond_to?(:validator)
+
+      if fields
+        return false if fields[:validator] && error.validator != fields[:validator]
+        return false if fields[:name]      && error.name      != fields[:name]
+      end
+
+      true
+    end
+
+    chain :with, :fields
+  end
+
```

Comment:
a_validate_error required some reading to understand, but is very clean, and that seems correct for gem code. 👍

Reply:
Yeah, I rather expect it will improve in time, but I felt like I had spent enough time on it for this work chunk.